fdbserver/ptxn/test/TestTLogPeek.actor.cpp

```diff
@@ -0,0 +1,390 @@
+		// PeekCursorBase::iterator. In this implementation we use iterators since iterators are implemented using the
+		// methods above.
+
+		for (const ptxn::VersionSubsequenceMutation& mutation : *pCursor) {
+			ASSERT(mutation == pContext->mutations[storageTeamID][index++]);
+		}
+	}
+
+	std::cout << "TestServerTeamPeekCursor"
+	          << ">> Checked " << index << " mutations out of " << pContext->mutations[storageTeamID].size() << " mutations."
+	          << std::endl;
+	ASSERT_EQ(index, pContext->mutations[storageTeamID].size());
+
+	return Void();
+}
+
+TEST_CASE("/fdbserver/ptxn/test/tLogPeek/cursor/MergedPeekCursor/base") {
+	state ptxn::test::TestTLogPeekOptions options(params);
+	state std::shared_ptr<ptxn::test::FakeTLogContext> pContext = std::make_shared<ptxn::test::FakeTLogContext>();
+	state Future<Void> tLog;
+	state std::vector<std::unique_ptr<ptxn::PeekCursorBase>> cursorPtrs;
+
+	// Limit the size of reply, to force multiple peeks
+	pContext->maxBytesPerPeek = params.getInt("maxBytesPerPeek").orDefault(32 * 1024);
+	ptxn::test::fillTLogWithRandomMutations(pContext, options.initialVersion, options.numMutations, options.numStorageTeams);
+	tLog = ptxn::test::initializeTLogForPeekTest(ptxn::MessageTransferModel::TLogActivelyPush, pContext);
+
+	for (auto storageTeamID : pContext->storageTeamIDs) {
+		cursorPtrs.emplace_back(
+		    new ptxn::StorageTeamPeekCursor(options.initialVersion, storageTeamID, pContext->pTLogInterface.get()));
+	}
+
+	state std::shared_ptr<ptxn::MergedPeekCursor> pMergedCursor =
+	    std::make_shared<ptxn::MergedPeekCursor>(cursorPtrs.begin(), cursorPtrs.end());
+	state size_t index = 0;
+	loop {
+		state bool remoteDataAvailable = wait(pMergedCursor->remoteMoreAvailable());
+		if (!remoteDataAvailable) {
+			std::cout << "TestMergedPeekCursor"
+			          << ">> TLog reported no more mutations available." << std::endl;
+			break;
+		}
+
+		for (const ptxn::VersionSubsequenceMutation& mutation : *pMergedCursor) {
+			ASSERT(mutation == pContext->allMutations[index++]);
+		}
+	}
+
+	std::cout << "TestServerTeamPeekCursor"
+	          << ">> Checked " << index << " mutations out of " << pContext->allMutations.size() << " mutations."
+	          << std::endl;
+	ASSERT_EQ(index, pContext->allMutations.size());
+
+	return Void();
+}
+
+namespace ptxn::test {
+
+namespace {
+
+const Version TEST_INITIAL_VERSION = 1000;
+const int NUM_VERSIONS = 4;
+const Version STEP_VERSION = 5;
+const int NUM_MUTATIONS_PER_VERSION = 8;
+const int NUM_TEAMS = 4;
+
+// Fill the TLog with the mutations:
+//
+// Version                1000          1005          1010          1015
+// Team 0 Subseq          1, 5          1, 5          1, 5          1, 5
+// Team 1 Subseq          2, 6          2, 6          2, 6          2, 6
+// Team 2 Subseq          3, 7          3, 7          3, 7          3, 7
+// Team 3 Subseq          4, 8          4, 8          4, 8          4, 8
```

Comment:
For better coverage, each version can have a random number of mutations, e.g., between 10-100, and each mutation is randomly assigned to a team.

Reply:
This is more used to ensure the cursor movement/iteration work properly. Using a hard-coded set of mutations is easier to implement. I think we can put more effort on this later?

Reply:
Sure. We can add such a test in another PR.